python/pyarrow/table.pxi

```diff
@@ -1638,26 +1638,48 @@ def table(data, names=None, schema=None, metadata=None):
             "Expected pandas DataFrame, python dictionary or list of arrays")
 
 
-def concat_tables(tables):
+def concat_tables(tables, with_promotion=False, MemoryPool memory_pool=None):
```

Comment:
Small nit: I might call this argument `promote=False` myself, but either way is okay.

Do you have thoughts about what the API might look like if we permitted safe casts?

Reply:
Do "safe casts" mean copy-free casts?

Since we agreed on having an options object (probably two options object, one for concat, the other one for UnifySchema), the casting can be an option in UnifySchemaOptions?

Reply:
Changed to `promote`.

Reply:
"Safe" casts are ones with unambiguous / lossless conversions. For example converting from int32 to int64 is safe but int64 to int32 is unsafe. Our cast functions when `safe=True` will return errors when data loss would occur